tests/pr_test.go

```diff
@@ -61,9 +65,13 @@ func setupOptions(t *testing.T, prefix string, dir string) *testhelper.TestOptio
 		Prefix:        prefix,
 		ResourceGroup: resourceGroup,
 	})
+	selectedRegion := validRegions[rand.Intn(len(validRegions))] // KMS encryption is only supported with the 'lakehouse-enterprise' plan. For regions 'au-syd' or 'ca-tor', the plan will switched to 'lakehouse-enterprise-mcsp'.
+	if selectedRegion == "au-syd" || selectedRegion == "ca-tor" {
```

Comment:
Why this workaround is required? Are you not handling the condition in root module to provision in au-syd or ca-tor without KMS if plan is `lakehouse-enterprise-mcsp` ?
Please recheck once.

Reply:
If the root module already enforces the KMS restriction for `lakehouse-enterprise-mcsp` and `lite` plans, then adding a workaround elsewhere is redundant. Why would it not make sense to ignore KMS values if they are explicitly blocked at the root? 
All enforcement should remain at the root to avoid logic duplication. 
Please re-evaluate the need for this workaround and discuss for clarity.